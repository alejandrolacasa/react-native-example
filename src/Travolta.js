import * as React from 'react';
import { NavigationScreenProp, NavigationState } from 'react-navigation';
import { Button } from 'react-native';
import Face from './Face/Face';

import travoltaSrc from './travolta.jpg';

type Props = {
  navigation: NavigationScreenProp<NavigationState, *>,
};

const Travolta = ({ navigation }: Props) => (
  <Face
    faceSrc={travoltaSrc}
    button={
      <Button title="Go back to Cage" onPress={() => navigation.goBack()} />
    }
  />
);

Travolta.navigationOptions = {
  title: 'Travolta',
  headerStyle: {
    backgroundColor: 'red',
  },
};

export default Travolta;
