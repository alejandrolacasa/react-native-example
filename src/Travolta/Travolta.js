import * as React from 'react';
import { NavigationScreenProp, NavigationState } from 'react-navigation';
import { Button } from 'react-native';
import Face from '../Face/Face';

import travoltaSrc from './travolta.jpg';

type Props = {
  navigation: NavigationScreenProp<NavigationState, *>,
};

const Travolta = ({ navigation }: Props) => (
  <Face
    button={
      <Button
        title="Go back to Cage"
        onPress={() => navigation.goBack()}
      />
    }
    faceSrc={travoltaSrc}
  />
);

Travolta.navigationOptions = {
  title: 'Travolta',
  headerStyle: {
    backgroundColor: 'red',
  },
};

export default Travolta;
