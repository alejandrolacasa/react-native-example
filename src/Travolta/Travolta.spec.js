import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Travolta from './Travolta';

describe('Travolta component', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Travolta />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call navigation on press', () => {
    const navigationMock = {
      goBack: jest.fn(),
    };

    const wrapper = shallow(<Travolta navigation={navigationMock} />);
    const button = shallow(wrapper.props().button);

    button.simulate('press');

    expect(navigationMock.goBack).toHaveBeenCalled();
  });
});
