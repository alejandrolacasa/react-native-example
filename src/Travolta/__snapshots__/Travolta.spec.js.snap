// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travolta component should render without crashing 1`] = `
<Face
  button={
    <Button
      onPress={[Function]}
      title="Go back to Cage"
    />
  }
  faceSrc={
    Object {
      "testUri": "../../../src/Travolta/travolta.jpg",
    }
  }
/>
`;
